spec/models/category_spec.rb

```diff
@@ -61,6 +61,32 @@
       end
     end
 
+    context 'for pileid-source Category' do
+      let(:category) { create(:category, name: 28301, source: 'pileid') }
+      let(:course) { create(:course, id: 28301) }
```

Comment:
This doesn't seem necessary. Setting the ID on the `Course` to be the same as the pileid is potentially going to cause confusion, and the course ID isn't relevant to what you're testing.